main.cpp

```diff
@@ -1,95 +1,252 @@
+        }
+        void setValvePositions(int v1, int v2, int v3)
+        {
+            moveValve1(v1);
+            moveValve2(v2);
+            moveValve3(v3);
+        }
 
-/*
- 3)
- */
+    };  
 
-/*
- 4)
- */
+    int slot = 2; 
+    int noteLength = 4;
+    int tempoBPM = 100; 
 
-/*
- 5)
- */
+    void setNote(int s, int l, int t)
+    {
+        slot = s;
+        noteLength = l;
+        tempoBPM = t;
+    } 
+    // set slot number  
+    void setSlot();
 
-/*
- 6)
- */
+    // set note length
+    void setNoteLength();
 
-/*
- 7)
- */
+    // set tempo
+    void setTempo(); 
+
+    // play note with current valve and slot positions 
+    void playNote();
+
+}; // end struct MechanicalTrumpet
+
+/* 
+2) Innerclock syncShift UDT
+*/ 
+// public members and member functions return values 
+struct SyncShift
+{
+    // modes are run=1, locked=2, bypass=3
+    int mode = 0;
+    int knob1 = 0; 
+    int knob2 = 0;
+    int knob3 = 0;
+    // knob4 is shift offset
+    double knob4 = 0;
+
+    ///////////////////////////////////////////////////////////
+    //
+    // setknobs 1 - 3 function calls return new knob value   
+    //
+    ///////////////////////////////////////////////////////////
+
+    int setKnob1(int k1) { knob1 = k1; return knob1; } 
+    int setKnob2(int k2) { knob2 = k2; return knob2; } 
+    int setKnob3(int k3) { knob1 = k3; return knob3; } 
+
+    ///////////////////////////////////////////////////////////
+    //
+    // setknob 4 function call return new knob value   
+    //
+    ///////////////////////////////////////////////////////////   
+
+    double setKnob4(double k4) { knob4 = k4; return knob4; }
+
+};  // end Struct SyncShift
+
+/* 
+3) mechanicalTrumpetV2 UDT (structures within a structure)
+*/
+
+    // secondary structure #1 
+struct ValvesV2
+{
+    int valve1 = 0;
+    int valve2 = 0;
+    int valve3 = 0;
+};  // end Struct Valves
+
+/* 
+4) SlotV2 UDT (embedded structure)
+*/
+struct SlotV2
+{
+    int slotNum = 0;
+    int slotNumLast = 0;
+};  // end Struct Slot
+
+/* 
+5) TimeV2 UDT (embedded structure)
+*/
+struct TimeV2
+{
+    int noteLength = 0;
+    int tempo = 0;
+};  // end Struct Time
+
+/* 
+6) MechanicalTrumpetV2 UDT (structures within structure)
+*/
+struct MechanicalTrumpetV2       
+{
+    ValvesV2 valves;
+    SlotV2 slot;
+    TimeV2 time;
+};  // end struct MechanicalTrumpetV2	
+
+/* 
+7) ElevatorDoors UDT (structure embedded two levels)
+*/
+
+struct ElevatorDoors
+{
+    void closeDoors(); 
+    void openDoors();  
+    bool doorsOpen = 1;
+    bool doorsClosed = 0;
+    bool doorsLocked = 0;
+    bool doorsUnlocked = 0;
+};	
+
+/* 
+8) ElevatorFloor UDT (structure embedded two levels)
+*/
+struct ElevatorFloor
+{
+    int previousFloor = 0;
+    int currentFloor = 0;
+    int nextFloor = 0;
+};
+/* 
+9) ElevatorCar UDT (embedded structure has embedded structures)
+*/
+struct ElevatorCar
+{ 
+    ElevatorDoors doors;    // embedded structure 
```

Comment:
Add a member function that opens the doors. 

- [ ] 3) add a couple member functions.